tests/phpunit/api/v3/PaymentTest.php

```diff
@@ -406,4 +406,61 @@ public function testUpdatePayment() {
     ));
   }
 
+  /**
+   * Test create payment api for paylater contribution
+   */
+  public function testCreatePaymentPayLater() {
+    $contributionParams = array(
+      'total_amount' => 100,
+      'currency' => 'USD',
+      'contact_id' => $this->_individualId,
+      'financial_type_id' => 1,
+      'contribution_status_id' => 2,
+      'is_pay_later' => 1,
+    );
+    $contribution = $this->callAPISuccess('Contribution', 'create', $contributionParams);
+
+    //Create partial payment
+    $params = array(
+      'contribution_id' => $contribution['id'],
+      'total_amount' => 100,
+    );
+    $payment = $this->callAPIAndDocument('Payment', 'create', $params, __FUNCTION__, __FILE__);
```

Comment:
If that's in response to my comment - the alternative to callAPIAndDocument is callAPISuccess - the former generates an example, visible in the api explorer - although you kind of have to run the tests without the constant to suppress example generation & commit the example. But it's better to use the latter function unless you want to generate an example.

Have a go running the tests locally with

 define('DONT_DOCUMENT_TEST_CONFIG', FALSE);

in your settings file - you will see what I mean